src/controllers/blog.controller.js

```diff
@@ -1,64 +1,139 @@
-import postData from '../models/post-data.json';
-import uniqid from 'uniqid';
 import jwt from 'jsonwebtoken';
-import getTokenAsFunc from '../middlewares/getTokenAsFunc';
 import dotEnv from 'dotenv';
+import mongoose from 'mongoose';
+const Post = require('../models/posts');
+
 
 dotEnv.config();
 
-// Assign each user a unique id
 
-let posts = [];
+export const displayPreviews = (req, res) => {
+    let postsLimit = 3;
+
+   let previewPosts = [];
+
+    Post.find()
+        .sort({ _id: -1 })
+        .limit(postsLimit)
+        .exec((err, previewPosts) => {
+            if (err) {
+                return res.status(500).json({
+                    status: "Server Error",
+                    message: "Cannot fetch the posts from the database"
+                })
+            }
 
-for (let post of postData) {
-    post = { id: uniqid('postid-'), ...post };
-    jwt.sign(post, process.env.SECRET_KEY, (err, token) => {
-        if (err)
-            return res.status(501).json({
-                status: 'Internal Server Error',
-                message: 'Cannot generate token for each post'
-            });
-        if (token) {
-            post = {postToken: token, ...post};
-            posts.push(post);
-        }
-    });
+            if (previewPosts.length !== 0) {
+                jwt.verify(req.token, process.env.SECRET_KEY, (err, authUser) => {
+                    if (err) {
+                        return res.status(200).json({
+                            status: 'Success - user not logged in',
+                            role: process.env.GUEST_USER_ROLE,
+                            userToken: null,
+                            previewPosts
+                        });
+                    }
+
+                    if (authUser.role === process.env.ADMIN_USER_ROLE) {
+                        return res.status(200).json({
+                            status: 'Success',
+                            role: process.env.ADMIN_USER_ROLE,
+                            userToken: req.token || null,
+                            previewPosts
+                        });
+                    }
+
+                    if (authUser.role === process.env.NORMAL_USER_ROLE) {
+                        return res.status(200).json({
+                            status: 'Success',
+                            role: process.env.NORMAL_USER_ROLE,
+                            userToken: req.token || null,
+                            previewPosts
+                        });
+                    }
+                });
+            } else {
+                return res.status(200).json({
+                    status: "Success",
+                    message: "No documents found in the database"
+                });
```

Comment:
is it necessary to store user roles in the environment?

Reply:
I saw it might be a security improvement that is why I added them in the environment

Reply:
I noted the remarks